mocks/mocks.py

```diff
@@ -18,15 +18,172 @@
+    # luminosity distance.
+    config['galaxy_distance'] = kwargs.get('galaxy_distance',
+                                           config['galaxy_distance'])
+    config['galaxy_distance'] = ensure_unit(config['galaxy_distance'], u.Mpc)
+    # The position of the target galaxy along the projection direction in the
+    # simulation box. Since it is coming from simulations, it is co-moving
+    # distance in Mpc.
+    config['target_galaxy_comoving_depth'] =\
+        kwargs.get('target_galaxy_comoving_depth',
+                   config['target_galaxy_comoving_depth'])
+    config['target_galaxy_comoving_depth'] = ensure_unit(
+        config['target_galaxy_comoving_depth'], u.Mpc)
+    config['viewing_axis'] = kwargs.get('viewing_axis',
+                                        config['viewing_axis'])
+    # The resolution of the simulation(parsec / pixel).
+    config['sim_pc_pixel'] = kwargs.get('sim_pc_pixel',
+                                        config['sim_pc_pixel'])
+    config['sim_pc_pixel'] = ensure_unit(config['sim_pc_pixel'],
+                                         u.pc / u.pixel)
+    config['particle_baryonic_mass_sim'] =\
+        kwargs.get('particle_baryonic_mass_sim',
+                   config['particle_baryonic_mass_sim'])
+    # To Do: The Hubble constant in the simulations is h, which is H / 100, it
+    # should be checked again that what constant is used in where and make
+    # proper conversion where it applies.
+    try:
+        config['hubble_constant'] = kwargs.get('hubble_constant',
+                                               config['hubble_constant'])
+    except KeyError:
+        config['hubble_constant'] = default_cosmology.H0
+    try:
+        assert config['hubble_constant']
+    except AssertionError:
+        config['hubble_constant'] = default_cosmology.H0
+    config['hubble_constant'] = ensure_unit(config['hubble_constant'],
+                                            u.km / (u.Mpc * u.s))
+    if config['hubble_constant'].value < 1:
+        config['hubble_constant'] = config['hubble_constant'] * 100
+    config['hubble_constant'] = ensure_unit(config['hubble_constant'],
+                                            u.km / (u.Mpc * u.s))
+    # Density of matter(Dark + baryonic)
+    try:
+        config['Omega_0'] = kwargs.get('Omega_0',
+                                       config['Omega_0'])
+    except KeyError:
+        config['Omega_0'] = default_cosmology.Om0
+    try:
+        assert config['Omega_0']
+    except AssertionError:
+        config['Omega_0'] = default_cosmology.Om0
+    # Density of baryonic matter.
+    try:
+        config['Omega_b'] = kwargs.get('Omega_b',
+                                       config['Omega_b'])
+    except KeyError:
+        config['Omega_b'] = default_cosmology.Ob0
+    try:
+        assert config['Omega_b']
+    except AssertionError:
+        config['Omega_b'] = default_cosmology.Ob0
```

Comment:
I think maybe I should change the lines that are parsing baryonic matter as follow:
```
    try:
        config['Omega_b'] = kwargs.get('Omega_b',
                                       config['Omega_b'])
    except KeyError:
        config['Omega_b'] = default_cosmology.Ob0
    try:
        assert config['Omega_b']
    except AssertionError:
        if config['Omega_b'] is not 0:
            config['Omega_b'] = default_cosmology.Ob0
```
So the code could work with Dark Matter Only simulations as well.

Reply:
As a general point, if you want to distinguish between something being `None`, or `0`, or some other value which evaluates to `False` (e.g. an empty `list`) then you can test for those directly, e.g. `if config['Omega_b'] is None:` or `if config['Omega_b'] == 0:`, etc.

As I noted in comments above I really don't think you need to allow for keys without values in the config file so you don't need this added complexity at all.